vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/main/java/com/vaadin/flow/component/spreadsheet/Spreadsheet.java

```diff
@@ -1965,9 +1967,9 @@ public void deleteSheetWithPOIIndex(int poiSheetIndex)
      *             index is invalid.
      */
     public void deleteSheet(int sheetIndex) throws IllegalArgumentException {
-        if (getNumberOfVisibleSheets() < 2) {
+        if (sheetIndex < 0 || sheetIndex >= getNumberOfVisibleSheets()) {
             throw new IllegalArgumentException(
-                    "A workbook must contain at least one visible worksheet");
+                    "Invalid index for visible sheet given.");
```

Comment:
This method was missing the checks for invalid sheetIndex.

Removed checks for deleting the last visible worksheet since it's also checked in `deleteSheetWithPOIIndex`